pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -226,12 +226,13 @@ func (m *kubeGenericRuntimeManager) startContainer(podSandboxID string, podSandb
 		}
 		msg, handlerErr := m.runner.Run(kubeContainerID, pod, container, container.Lifecycle.PostStart)
 		if handlerErr != nil {
-			m.recordContainerEvent(pod, container, kubeContainerID.ID, v1.EventTypeWarning, events.FailedPostStartHook, msg)
+			s, _ := grpcstatus.FromError(handlerErr)
```

Comment:
can you explain why it is needed here? Message returned by handlers is quite elaborative 
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/lifecycle/handlers.go#L59
and every handler might not be a grpc call.

Reply:
I added it here only because this handler error will be missing if the `killContainer` below succeeds.

Reply:
handle error will be missing from where? log?
sorry if I am taking time to understand.

Reply:
line 231 will log handleErr. However， kubelet will log the err only if killContainer failed.

Reply:
Can we add an additional log in `else` here?https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/kuberuntime_container.go#L287
 
Also it looks weird in the below log that `err` from `killContainer` is not getting logged. I think It should print that too.
https://github.com/kubernetes/kubernetes/blob/1d38084930b56be9b5830c20bd5b464b1f955047/pkg/kubelet/kuberuntime/kuberuntime_container.go#L284-L285

Reply:
Updated, add the err to the log as well. And print the log when killing container success as well.